recipes/mbits-lngs/all/conanfile.py

```diff
@@ -0,0 +1,189 @@
+        tc.variables["LNGS_LINKED_RESOURCES"] = True
+        tc.variables["LNGS_NO_PKG_CONFIG"] = True
+        if (
+            self.settings.os != self._settings_build.os
+            or self.settings.arch != self._settings_build.arch
+        ):
+            tc.variables["LNGS_REBUILD_RESOURCES"] = False
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    @property
+    def _cmake_install_base_path(self):
+        return os.path.join("lib", "cmake")
+
+    def package(self):
+        copy(
+            self,
+            pattern="LICENSE",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+        )
+        cmake = CMake(self)
+        cmake.install()
+
+        os.unlink(
+            os.path.join(
+                self.package_folder, self._cmake_install_base_path, "mbits-lngs.cmake"
+            )
+        )
+        os.unlink(
+            os.path.join(
+                self.package_folder,
+                self._cmake_install_base_path,
+                "mbits-lngs-{}.cmake".format(str(self.settings.build_type).lower()),
+            )
+        )
+        with open(
+            os.path.join(
+                self.package_folder,
+                self._cmake_install_base_path,
+                "module-mbits-lngs.cmake",
+            ),
+            "w",
+            encoding="UTF-8",
+        ) as cfg:
+            exe_ext = ".exe" if self.settings.os == "Windows" else ""
+            lngs_filename = "lngs" + exe_ext
+            module_folder_depth = len(
+                os.path.normpath(self._cmake_install_base_path).split(os.path.sep)
+            )
+            lngs_rel_path = "{}bin/{}".format(
+                "".join(["../"] * module_folder_depth), lngs_filename
+            )
+            cfg.write(
+                textwrap.dedent(
+                    """\
+                if(NOT TARGET mbits::lngs)
+                    if(CMAKE_CROSSCOMPILING)
+                        find_program(LNGS_PROGRAM lngs PATHS ENV PATH NO_DEFAULT_PATH)
+                    endif()
+                    if(NOT LNGS_PROGRAM)
+                        set(LNGS_REL_PATH "{lngs_rel_path}")
+                        set(LNGS_PROGRAM "${{CMAKE_CURRENT_LIST_DIR}}/${{LNGS_REL_PATH}}")
+                    endif()
+                    get_filename_component(LNGS_PROGRAM "${{LNGS_PROGRAM}}" ABSOLUTE)
+                    set(Mbitslngs_LNGS_EXECUTABLE ${{LNGS_PROGRAM}} CACHE FILEPATH "The lngs tool")
+                    add_executable(mbits::lngs IMPORTED)
+                    set_property(TARGET mbits::lngs PROPERTY IMPORTED_LOCATION ${{Mbitslngs_LNGS_EXECUTABLE}})
+                endif()
+            """.format(
+                        lngs_rel_path=lngs_rel_path
+                    )
+                )
+            )
```

Comment:
As I understand, this code might be required to perform the translations but I don't fully understand what is required and why you have to "patch" these cmake files. I think this requires some clarification in the recipe with some comments to understand it and t be able to keep it working in the future

Reply:
This is so that the `lngs` binary could be used through `COMMAND mbits::lngs` (see `recipes/mbits-lngs/all/test_package/CMakeLists.txt` for usage). This was inspired by recipe for `protobuf` in how they expose `protoc` message compiler.